tests/bsim/bluetooth/mesh/src/test_provision.c

```diff
@@ -1240,13 +1291,72 @@ static void test_device_pb_remote_server_proved(void)
 	PASS();
 }
 
+static void reprovision_remote_devkey_server(const uint16_t initial_addr)
+{
+	struct bt_mesh_key prev_dev_key;
+
+	memcpy(&prev_dev_key, &bt_mesh.dev_key, sizeof(struct bt_mesh_key));
+
+	ASSERT_OK(k_sem_take(&reprov_sem, K_SECONDS(30)));
+	ASSERT_EQUAL(initial_addr, bt_mesh_primary_addr());
+
+	/* Let Configuration Client activate the new Device Key and verify that it has
+	* been changed.
+	*/
+	k_sleep(K_SECONDS(2));
+	ASSERT_TRUE(memcmp(&prev_dev_key, &bt_mesh.dev_key, sizeof(struct bt_mesh_key)));
```

Comment:
I do not think the comparison of keys in such a manner is correct. You do not know which key value is hidden behind key_id in PSA. The key_id might be destroyed and allocated\imported again during provisioning or other algorithms. After this, the absolutely different key will have the same key_id.

you have to export the raw key value at the beginning of the test and in the end of the test and compare the raw values, not key_ids everywhere in the tests you did this checking.

I guess this was my mistake when I added support PSA in bsim tests. :(